content/articles/building-an-ensemble-learning-based-regression-model-using-python/index.md

```diff
@@ -0,0 +1,173 @@
+submission = pd.read_csv('/kaggle/input/Dataset2/sample_submission.csv')
+```
+#### Feature Selection
+
+Here we shall select the useful features from our dataset and remove the not so useful/ impactful features. In this dataset, the not so useful features would be; `id`, `target`, and `kfold`.
+
+To select the useful features, run the following block of code:
+```python
+useful_features = [i for i in data.columns if i not in ("id", "target","kfold")]
+object_cols = [col for col in useful_features if "cat" in col]
+test_data = test_data[useful_features]
+```
+
+#### Modelling
+
+To build our model, we shall run the following block of code:
+```python
+final_predictions =[]
+
+for fold in range(5):
+    xtrain = data[data.kfold != fold].reset_index(drop=True)
+    xvalid = data[data.kfold == fold].reset_index(drop=True)
+    xtest = test_data.copy()
+    
+    ytrain = xtrain.target
+    yvalid = xvalid.target
+    
+    xtrain = xtrain[useful_features]
+    xvalid = xvalid[useful_features]
+    
+    # Data Encoding 
+    oe = OrdinalEncoder()
+    xtrain[object_cols] = oe.fit_transform(xtrain[object_cols])
+    xvalid[object_cols] = oe.transform(xvalid[object_cols])
+    xtest[object_cols] = oe.transform(xtest[object_cols])
+    
+    # Model Training
+    model = XGBRegressor(random_state = fold, n_jobs=5)
+    model.fit(xtrain, ytrain)
+    preds_valid = model.predict(xvalid)
+    preds_test = model.predict(xtest)
+    final_predictions.append(preds_test)
+    print(fold, mean_squared_error(yvalid, preds_valid, squared=False))
+```
+For each fold, we shall encode the data and then train the model using XGBoost (extreme Gradient Boosting) which is an ensemble learning technique to boost the performance of our model.
+XGBoost is a regularized boosting technique that provides high predictive power and is faster compared to other boosting techniques.
+We shall then evaluate each fold individually and print out the results of the model.
+ 
+ #### Model Evaluation
+  
+ After individually evaluating each fold, we shall now evaluate our model's performance by getting mean of the predictions on our test data. To do this, use the following code block:
+ 
+```python
+preds = np.mean(np.column_stack(final_predictions), axis=1)
+```
+To see how our model performed, we shall output the results of our model's prediction using the following code:
+```python
+submission.target =preds
+submission.to_csv("submission1.csv", index=False)
+``` 
+>**Bonus:** You can submit a late submission to 30 days ML Kaggle challenge and see how your model peforms i.e If you had signed up for the challenge earlier.
+
+### Hyperparameter optimization
+
+In this process, we'll be fine-tuning and optimizing our model's algorithm paremeters until we achieve the desired result.
+A few common XGBoost parameters with a large effect on the model perfomance include; **n_jobs**, **max_depth**, **learning_rate**, **n_estimators**, **colsample_bytree**, and **subsample**.
+ 
```

Comment:
I can't see where you've fine-tuned and optimized the model. This section seems incomplete.